### PR TITLE
Individual Assessment get new permission amend finalized record

### PR DESCRIPTION
**Pull request by shecken**
With this PR the feature [Amend Finalized Records](https://www.ilias.de/docu/goto.php?target=wiki_1357_New_Permission%3A_Amend_Finalized_Records) will be implemented.

### CODE REVIEW

File: Modules/IndividualAssessment/classes/class.ilIndividualAssessmentMembersGUI.php [hunk 1]
```diff
@@ -28,13 +28,14 @@ public function __construct($a_parent_gui, $a_ref_id) {
 		$this->tpl =  $DIC['tpl'];
 		$this->lng = $DIC['lng'];
 		$this->toolbar = $DIC['ilToolbar'];
-		$this->access_handler = $this->object->accessHandler();
+		$this->iass_access = $this->object->accessHandler();
```
Comment (klees) [inline Modules/IndividualAssessment/classes/class.ilIndividualAssessmentMembersGUI.php:31]: Why did you rename this?

File: Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php [hunk 2]
```diff
@@ -72,4 +83,135 @@ protected function getRoleTitleByObj(ilObjIndividualAssessment $iass) {
...
+	 * @param string 	$operation
+	 *
+	 * @return bool
+	 */
+	protected function cacheCheckAccessToObj($operation)
+	{
+		$iass_id = $this->iass->getId();
+		$user_id = $this->usr->getId();
+
+		if (!isset($this->mass_global_permissions_cache[$iass_id][$user_id][$operation])) {
```
Comment (klees) [inline Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php:216]: Please declare `mass_global_permissions_cache`.

File: Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php [hunk 3]
```diff
@@ -72,4 +83,135 @@ protected function getRoleTitleByObj(ilObjIndividualAssessment $iass) {
...
 	}
+
+	/**
+	 * User view iass
+	 *
+	 * @param bool 	$use_cache
+	 *
+	 * @return bool
+	 */
+	public function mayViewIass($use_cache = true)
```
Comment (klees) [inline Modules/IndividualAssessment/classes/AccessControl/class.ilIndividualAssessmentAccessHandler.php]: Remove that `Iass`-suffix please. If you feel it is necessary to distinguish that method from `mayViewUser` by a suffix, use `Object`.

File: Modules/IndividualAssessment/classes/class.ilIndividualAssessmentMemberGUI.php [hunk 4]
```diff
@@ -248,8 +312,180 @@ protected function fillForm(ilPropertyFormGUI $a_form, ilIndividualAssessmentMem
...
+
+		return false;
+	}
+
+	/**
+	 * Current user has permission to edit learning progess
+	 *
+	 * @return bool
+	 */
+	protected function userCanGrade()
```
Comment (klees) [inline Modules/IndividualAssessment/classes/class.ilIndividualAssessmentMemberGUI.php]: Why do you use `may` sometimes and `can` some other times?

**Review by klees — CHANGES_REQUESTED** [review]
LGTM in general, thx for boyscouting. Please solve the naming issues.

File: Modules/IndividualAssessment/classes/class.ilIndividualAssessmentMembersGUI.php (hunk 1 above)
Comment (shecken) [reply Modules/IndividualAssessment/classes/class.ilIndividualAssessmentMembersGUI.php:31]: iass has his own access handler class. Rename makes clear this is not the global ilias access handler.

**Comment by klees** [comment]
@shecken Gna, please update once again.

**Pull request merged**